projects/label-designer/src/lib/label-designer.helper.ts

```diff
@@ -42,4 +42,19 @@ export class LabelDesignerHelper {
       }
     });
   }
+
+  static fieldsAreSame(fields1: ILabelField[], fields2: ILabelField[]): boolean {
+    if (fields1.length !== fields2.length) {
+      return false;
+    }
+    const keys = new Set<string>();
+    fields2.forEach(field => keys.add(field.field));
+    let result = true;
+    fields1.forEach(field => {
+      if (result && !keys.has(field.field)) {
+        result = false;
+      }
+    });
+    return result;
```

Comment:
Could be more simple with declarative approach, e.g.:

```
return fields1.every(f=> keys[f.field]))
```

That would also prevent excessive iteration, as the `every` would stop when a non matching item is found.

Continuing with declarative style, the `keys` could also be declared from `fields2` using reduce. IMO declarative style is more readable than line-by-line code execution. But that's just my opinion, take it as an idea instead of a change requirement :)

**EDIT:** I just noticed that this code block was copy-pasted from elsewhere, so no need to change anything as this isn't new code. I'll leave this comment here anyways. If you have time, you can improve the code of course!